hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/test/java/org/apache/hadoop/yarn/server/router/webapp/TestRouterWebServicesREST.java

```diff
@@ -198,6 +202,7 @@ public Boolean get() {
         }
       }, 1000, 10 * 1000);
     } catch (Exception e) {
+      LOG.error("Web app not running", e);
```

Comment:
We should also add some more information, because the current log is not enough for analysis after an error is reported.

The error stack is as follows:

```
Error Message
Web app not running
Stacktrace
java.lang.AssertionError: Web app not running
	at org.junit.Assert.fail(Assert.java:89)
	at org.apache.hadoop.yarn.server.router.webapp.TestRouterWebServicesREST.waitWebAppRunning(TestRouterWebServicesREST.java:201)
	at org.apache.hadoop.yarn.server.router.webapp.TestRouterWebServicesREST.setUp(TestRouterWebServicesREST.java:225)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
....
```

We can also view the following test report
[Test-Report](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-4843/12/testReport/org.apache.hadoop.yarn.server.router.webapp/TestRouterWebServicesREST/org_apache_hadoop_yarn_server_router_webapp_TestRouterWebServicesREST/)

Just looking at the error message of the stack, it seems impossible to analyze why the `Web app is not running`.

Reply:
> Why do we do this? Isn't it clear enough with the "fail()" message?

I understand your thoughts, the  "fail()" message is enough, generally it is caused by timeout, I increased the timeout.